ev11l78a: add en_sink as a custom binding.

Flameeyes opened the pull request:
ev11l78a: add en_sink as a custom binding.

This exposes the EN_SINK signal as a custom binding, since it is ore to
the business logic of the board.

EN_SINK is a GPIO line connected to a MOSFET that provides power on the
screw terminal of the board, and should be enabled after USB-PD
negotiation.

Signed-off-by: Diego Elio Pettenò <flameeyes@meta.com>

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/zephyrproject-rtos/zephyr/pull/61740).
* __->__ #61740
* #62478
* #61797

gmarull commented on boards/arm/ev11l78a/ev11l78a.dts:
```diff
@@ -41,6 +54,11 @@
 		sense-gain-mult = <100>;
 	};
 
+	enable_sink: enable_sink {
+		compatible = "microchip,ev11l78a-ensink";
+		gpios = <&porta 22 GPIO_ACTIVE_HIGH>;
+	};
```
looks like a `regulator-fixed`?

gmarull replied:
so if controlled by TCPC it should be a property there, something like:

```
tcpc {
    enable-sink-gpios =  <>;
};
```

then TCPC code would use the GPIO API. Or, if using regulators:

```
enable_sink: enable-sink {
    compatible = "regulator-fixed";
    enable-gpios = <...>;
};

...

tcpc {
    ...
    sink-supply = <&enable_sink>;
};
```

Then TCPC would use the regulators API (main advantage is that is reference counted, useful if supply is shared).

gmarull replied:
It depends on how it is used, if by multiple clients then regulator is a good fit, is by a single client, a GPIO does the job. A typical case for GPIO usage is e.g. on sensors that have a shutdown pin. I don't know this case in detail to suggest one way or the other.

Flameeyes replied:
To be honest, this is almost the literal case from [the documentation you linked earlier](https://docs.zephyrproject.org/latest/samples/basic/custom_dts_binding/README.html) as it's a GPIO connected to a specific MOSFET…

Leaving it as a generic GPIO doesn't seem right though — it's most certainly no longer "General-purpose" once it's wired to the MOSFET on the board, and it has a very specific function.

Flameeyes commented on boards/arm/ev11l78a/ev11l78a.dts:
```diff
@@ -32,6 +33,18 @@
 			gpios = <&porta 23 GPIO_ACTIVE_HIGH>;
 			label = "CAP_MIS";
 		};
+
+		// The following are not LEDs on the board, but rather
+		// testpoints. On the other hand, they are "fit the part."
+		ind_1_5a: led_1 {
+			gpios = <&porta 1 GPIO_ACTIVE_HIGH>;
+			label = "1.5A_IND";
+		};
+
+		ind_3_0a: led_2 {
+			gpios = <&porta 15 GPIO_ACTIVE_HIGH>;
+			label = "3.0A_IND";
+		};
 	};
```
It seemed the easiest way to expose these with an actual label — but I'm not convinced myself. What other options are there?

Flameeyes reviewed (commented):
Let me move the important change (memory and flash size) to #61744 that is more generic, and I'll follow up here about the regulator/sink enable option.

nandojve commented:
You need rebase : )

nandojve commented:
Ping

Flameeyes committed:
ev11l78a: add en_sink as a custom binding.

This exposes the EN_SINK signal as a custom binding, since it is ore to
the business logic of the board.

EN_SINK is a GPIO line connected to a MOSFET that provides power on the
screw terminal of the board, and should be enabled after USB-PD
negotiation.

Signed-off-by: Diego Elio Pettenò <flameeyes@meta.com>

Flameeyes committed:
ev11l78a: configure DAC peripheral

This configures the DAC in the same way as it is configured for the
arduino_zero and other similar boards.

It also includes the board in the DAC tests.

Signed-off-by: Diego Elio Pettenò <flameeyes@meta.com>

Flameeyes committed:
ev11l78a: minimize memory usage in the default config.

The ATSAMD20E16 used in this board only has 8KiB SRAM, so the defaults
are not really suitable for anything close to useful, particularly the
sink sample that most closely resembles the original demo for the board.

This should also allow more tests not to be skipped for this board as
otherwise they overflow RAM usage.

Signed-off-by: Diego Elio Pettenò <flameeyes@meta.com>

nandojve commented:
Hi @Flameeyes ,

Is there anything pending from your side on this?
